cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
 import importlib
 import importlib.util
 import inspect
+import os
+import subprocess
 import sys
 from pathlib import Path
 
 import click
 import typer
 
+from cognite.dm_clients.config import settings
 from cognite.dm_clients.domain_modeling.schema import Schema
 from cognite.gqlpygen.generator import to_client_sdk
 from cognite.gqlpygen.misc import to_client_name, to_schema_name
 
 app = typer.Typer()
 
 
-@app.command("topython", help="Input a .graphql schema to create pydantic schema.")
+_settings_cognite = settings.get("cognite", {})
+_cdf_cluster = _settings_cognite.get("cdf_cluster")
+_tenant_id = _settings_cognite.get("tenant_id")
+_client_id = _settings_cognite.get("client_id")
+_client_secret = _settings_cognite.get("client_secret")
+_project = _settings_cognite.get("project")
+
+_settings_dm_clients = settings.get("dm_clients", {})
+_space = _settings_dm_clients.get("space")
+_datamodel = _settings_dm_clients.get("datamodel")
+_schema_version = _settings_dm_clients.get("schema_version")
+
+_settings_local = settings.get("local", {})
+_prefix = _settings_local.get("prefix", "")
+_graphql_schema = _settings_local.get("graphql_schema")
+_schema_module = _settings_local.get("schema_module")
+
+_schema_dir = Path(_graphql_schema).parent if _graphql_schema else None
+
+_client_secret_none = "None (use device flow)"
+if _client_secret:
+    _client_secret_hidden = f"{_client_secret[:3]}{'*' * (len(_client_secret) - 3)}"
+else:
+    _client_secret_hidden = _client_secret_none
```

Comment:
If I am not mistaken `pydantic` should have some support for hiding the password. See `SecretStr` field https://docs.pydantic.dev/usage/types/#secret-types

Reply:
Nice, wasn't aware. But I think it is useful to see part of the secret (e.g. first 3 characters) to confirm which values is being loaded. I don't see support for this in `SecretStr`.

Reply:
Agree on the `[:3]`, but it shouldn't leak the length of the pw.